### ISSUE TITLE
Training Extension Causes Panic in UI code as a result of invalid progress rehydration

### ISSUE THREAD

**Issue by fuzing** [tracel-ai/burn#5091]
ref issue [5082](https://github.com/tracel-ai/burn/issues/5082), which fixed the device issue but there appears to be another problem for extending training, this time in the UI code.

If I train a/my model to, say, 3 epochs, everything is fine (reproducible for both backends I'm testing with, namely CUDA and torch-cuda).

However, if I train my model to 2 epochs and then extend the training to 3 epochs, I get the following error:
```
thread 'train-renderer' (475626) panicked at /home/some-folder/Projects/burn/crates/burn-train/src/renderer/tui/progress.rs:199:21:
attempt to subtract with overflow
```

The error consistently appears part way through the extension epoch.  I'm wondering if this is an issue with rehydration of progress state for continuation of training.

The panic is here:
```
        let epoch = progress.global.items_processed - starting_epoch;
```
where items_processed is 1, but the starting_epoch is 2, underflowing epoch (usize).

**Comment by fuzing** [comment]
Thanks @SAY-5 - I'll throw this out there for comment though........

Is performing a saturating_sub() the correct fix for this? (depends on current epoch semantics which I haven't checked).  Shouldn't the progress epoch be rehydrated to become "number of epochs previously completed" + 1.

Otherwise, if I extend a previously trained-for-5-epochs model for 1 epoch, won't progress then be saved as if only a single epoch has elapsed (and all the progress/history will be saved/checkpointed as if we're tackling epoch 1, instead of epoch 6).  Then, if I extend for another epoch, won't the progress state be in the wrong position after the first training extension?

(clearly, this is only be a problem if this progress is used by things like the checkpointing mechanism - otherwise, if only used for the train-renderer then this might be a moot point).

**Comment by fuzing** [comment]
@SAY-5 - I applied your PR locally and it appears to perform as expected, mostly.  Checkpoint information is written correctly as model-6.bpk/optim-6.bpk/scheduler-6.bpk.

The main serious caveat is that training and validation logs are overwritten as epoch-1, which is generally fine for my current use-case (depends upon one's view of "epoch" - is it for this training run, or overall).

This is, however, likely to be confusing/problematic to anyone consuming the logs who would be expecting results to appear in the same folder as the actual follow-on epoch namesake, i.e. in this case, within an epoch-6 folder.  Pytorch would use the equivalent of the epoch-6 folder and doesn't seem to clobber earlier epoch results.

It also clearly risks destroying/overwriting previous epoch training and validation logs, which an unsuspecting consumer might find troublesome (epoch-1 logs no longer align with checkpoint/model-1 results).  Personally, I'd prefer that all logs/checkpoints/etc. align to the same absolute epoch (irrespective of extension) so that full history is preserved and not overwritten.  My workflow for other projects has been:

- Train model for x epochs
- Follow-on train model for an additional y epochs
- Follow-on train model for an additional z epochs
- etc.

Lastly, pick "best" model based on criteria extracted from logs (which will be impossible because the first "y" and/or "z" logs have been overwritten).

Many thanks again for your quick response!

**Comment by Charles23R** [comment]
Fixed by #5116
